src/Data/LLVM/BitCode/Parse.hs

```diff
@@ -624,22 +660,22 @@ addFwdFNEntry :: Int -> Int -> ValueSymtab -> ValueSymtab
 addFwdFNEntry i o t = t { fnSymtab = IntMap.insert i (Right o) (fnSymtab t) }
 
 -- | Lookup the name of an entry. Returns @Nothing@ when it's not present.
-entryNameMb :: Int -> Parse (Maybe String)
+entryNameMb :: HasParseEnv m => Int -> m (Maybe String)
 entryNameMb n = do
-  symtab <- liftFinalize getValueSymtab
+  symtab <- getValueSymtab
   return $! fmap renderName
          $  IntMap.lookup n (valSymtab symtab) `mplus`
             IntMap.lookup n (fnSymtab symtab)
 
 -- | Lookup the name of an entry.
-entryName :: Int -> Parse String
+entryName :: HasParseEnv m => HasValueTable m => MonadFail m => Int -> m String
```

Comment:
Similarly, I have a slight preference towards tupling the constraints here.

Reply:
Tupled.